service/lib/dinstaller/dbus/storage/manager.rb

```diff
@@ -131,6 +123,77 @@ def calculate_proposal(dbus_settings)
           success ? 0 : 1
         end
 
+        dbus_interface PROPOSAL_CALCULATOR_INTERFACE do
+          dbus_reader :available_devices, "a(ssa{sv})"
+
+          dbus_reader :volume_templates, "aa{sv}"
+
+          dbus_reader :result, "o"
+
+          # result: 0 success; 1 error
+          dbus_method :Calculate, "in settings:a{sv}, out result:u" do |settings|
+            busy_while { calculate_proposal(settings) }
+          end
+        end
+
+        ISCSI_INITIATOR_INTERFACE = "org.opensuse.DInstaller.Storage1.ISCSI.Initiator"
+        private_constant :ISCSI_INITIATOR_INTERFACE
+
+        # Gets the iSCSI initiator name
+        #
+        # @return [String]
+        def initiator_name
+          backend.iscsi.initiator.name || ""
+        end
+
+        # Sets the iSCSI initiator name
+        #
+        # @param value [String]
+        def initiator_name=(value)
+          backend.iscsi.initiator.name = value
+        end
+
+        # Performs an iSCSI discovery
+        #
+        # @param address [String] IP address of the iSCSI server
+        # @param port [Integer] Port of the iSCSI server
+        # @param options [Hash<String, String>] Options from a D-Bus call:
+        #   @option Username [String] Username for authentication by target
+        #   @option Password [String] Password for authentication by target
+        #   @option ReverseUsername [String] Username for authentication by initiator
+        #   @option ReversePassword [String] Username for authentication by inititator
+        #
+        # @return [Integer] 0 on success, 1 on failure
+        def iscsi_discover(address, port, options = {})
+          success = backend.iscsi.discover_send_targets(address, port, iscsi_auth(options))
+          success ? 0 : 1
+        end
+
+        # Deletes an iSCSI node from the database
+        #
+        # @param path [::DBus::ObjectPath]
+        # @return [Integer] 0 on success, 1 on failure
```

Comment:
Ignorant question: is that the typical behavior on D-Bus APIs? I mean, doesn't it make sense to return different exit codes for the different problems instead of "1" in all cases?

Reply:
Yes, it makes sense to return different error code. Fixing it.